AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginIntegrationTests/DataStoreConfigurationTests.swift

```diff
@@ -0,0 +1,150 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+//
+//  DataStoreConfigurationTests.swift
+//  AWSDataStoreCategoryPluginIntegrationTests
+//
+//  Created by Guo, Rui on 6/15/20.
+//  Copyright © 2020 Amazon Web Services. All rights reserved.
+//
+
+import XCTest
+
+import AmplifyPlugins
+import AWSMobileClient
+import AWSPluginsCore
+
+@testable import Amplify
+@testable import AmplifyTestCommon
+@testable import AWSDataStoreCategoryPlugin
+
+class DataStoreConfigurationTests: XCTestCase {
+
+    func testConfigWithSameSchema() throws {
+
+        let databaseCreationDone = expectation(description: "Old database deleted and new database recreated")
+
+        do {
+            let dataStorePlugin = AWSDataStorePlugin(modelRegistration: AmplifyModels())
+            try Amplify.add(plugin: dataStorePlugin)
+            try Amplify.configure()
+            print("Amplify configured with DataStore plugin")
+        } catch {
+            print("Failed to initialize Amplify with \(error)")
+        }
+
+        let time: Temporal.DateTime = .now()
+        let post = Post(title: "title", content: "content", createdAt: time)
+        let id = post.id
+
+        Amplify.DataStore.save(post, completion: { result in
+            switch result {
+            case .success:
+                print("Post saved successfully!")
+                databaseCreationDone.fulfill()
+            case .failure(let error):
+                print("Error saving post \(error)")
+            }
+        })
+        wait(for: [databaseCreationDone], timeout: 10)
```

Comment:
nit: can probably still use `timeout: TestCommonConstants.networkTimeout` even if we know these tests are not making network calls, or store 10 seconds as a constant in class and reuse it, or maybe TestCommonConstants can have a regular one like `timeout = TimeInterval(10)` then you can use it across as `TestCommonConstants.timeout`

Reply:
Then let me just use networkTimeout